androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/fragments/BeerFragment.java

```diff
@@ -0,0 +1,190 @@
+ * to handle interaction events.
+ * Use the {@link BeerFragment#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class BeerFragment extends BaseFragment {
+
+    private ScrollView mScrollView;
+    private BeerViewModel viewModel;
+
+    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+    private static final String ARG_BEER_NAME = "beer_name";
+    private static final String ARG_BEER_STYLE = "beer_style";
+    private static final String ARG_BEER_DESCRIPTION = "beer_description";
+    private static final String ARG_BEER_ABV = "beer_abv";
+    private static final String ARG_BEER_IBUS = "beer_ibus";
+    private static final String ARG_BEER_SRM = "beer_srm";
+    private static final String ARG_BEER_PRICE = "beer_price";
+    private static final String ARG_BEER_REVIEW = "beer_review";
+    private static final String ARG_BEER_DESCRIPTION_IMAGE = "description_image";
+
+
+    public String name;
+    public String style;
+    public String description;
+    public String description_image;
+    public double price;
+    public double abv;
+    public int ibus;
+    public int srm;
+    public double review;
+
+    private OnFragmentInteractionListener mListener;
+
+    public BeerFragment() {
+        // Required empty public constructor
+    }
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     *
+     * @return A new instance of fragment BeerFragment.
+     */
+    // TODO: Rename and change types and number of parameters
+    public static BeerFragment newInstance(String beer_name,
+                                           String beer_style,
+                                           String beer_description,
+                                           String beer_description_image,
+                                           double beer_price,
+                                           double beer_abv,
+                                           int beer_ibus,
+                                           int beer_srm,
+                                           double beer_review) {
+        BeerFragment fragment = new BeerFragment();
+        Bundle args = new Bundle();
+        args.putString(ARG_BEER_NAME, beer_name);
+        args.putString(ARG_BEER_STYLE, beer_style);
+        args.putString(ARG_BEER_DESCRIPTION, beer_description);
+        args.putString(ARG_BEER_DESCRIPTION_IMAGE, beer_description_image);
+        args.putDouble(ARG_BEER_PRICE, beer_price);
+        args.putDouble(ARG_BEER_ABV, beer_abv);
+        args.putDouble(ARG_BEER_REVIEW, beer_review);
+        args.putInt(ARG_BEER_IBUS, beer_ibus);
+        args.putInt(ARG_BEER_SRM, beer_srm);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
```

Comment:
Of course you don't have to do it, I hope you don't have to deal with android code again, but just in case. For places where there are too many parameter sometimes is better to encapsulate them all in an object and make it implement [Parcelable](https://developer.android.com/reference/android/os/Parcelable.html) interface. That way it can be serialized and passed directly to fragments.

Reply:
.. Interesting .. thanks for the tip! it was a super pain to deal with all these args.